java/code/src/com/suse/manager/webui/controllers/SystemsController.java

```diff
@@ -170,6 +178,78 @@ private ModelAndView virtualListPage(Request requestIn, Response responseIn, Use
         data.put("is_admin", userIn.hasRole(RoleFactory.ORG_ADMIN));
         return new ModelAndView(data, "templates/systems/virtual-list.jade");
     }
+
+    /**
+     * Get the virtual systems list page
+     *
+     * @param request the request
+     * @param response the response
+     * @param user the user
+     * @param server the server
+     * @return the jade rendered template
+     */
+     public ModelAndView mgrServerInfoPage(Request request, Response response, User user, Server server) {
+        MgrServerInfo info = server.getMgrServerInfo();
+        Map<String, Object> data = new HashMap<>();
+
+        data.put("is_admin", user.hasRole(RoleFactory.ORG_ADMIN));
+        data.put("name", server.getName());
+        data.put("version", Optional.ofNullable(info.getVersion())
+                .map(PackageEvr::toUniversalEvrString)
+                .orElse(""));
+        data.put("reportDbName", info.getReportDbName());
+        data.put("reportDbHost", info.getReportDbHost());
+        data.put("reportDbPort", info.getReportDbPort());
+        data.put("reportDbUser", Optional.ofNullable(info.getReportDbCredentials())
+                .map(Credentials::getUsername)
+                .orElse(""));
+        data.put("reportDbLastSynced", info.getReportDbLastSynced());
+
+        return new ModelAndView(data, "templates/systems/mgr-server.jade");
+     }
+
+     /**
+      * Autogenerate a new password for a Mgr Servers report database,
+      * set it in the DB and schedule a password change on that server.
+      *
+      * @param request the request
+      * @param response the response
+      * @param user the user
+      * @return the json response
+      */
+     public String mgrServerNewReportDbPassword(Request request, Response response, User user) {
+         String sidStr = request.params("sid");
+         long sid;
+         try {
+             sid = Long.parseLong(sidStr);
+         }
+         catch (NumberFormatException e) {
+             LOG.error(String.format("SystemID (%s) not a long", sidStr));
+             return json(response, HttpStatus.SC_BAD_REQUEST, ResultJson.error("invalid_systemid"));
+         }
+         Server server = null;
+         try {
+             server = SystemManager.lookupByIdAndUser(sid, user);
+         }
+         catch (Exception e) {
+             LOG.error(e.getMessage());
```

Comment:
Nitpicking, but I would change this so the stacktrace of the exception is not lost:

```suggestion
             LOG.error(String.format("System (%s) cannot be retrieved", sid), e);
```

Reply:
The message we get from the exception is pretty good and the full stacktrace was not really helpfull.
So I decided to log only the message.